mersenne-31/src/mersenne_31.rs

```diff
@@ -90,6 +91,25 @@ impl Distribution<Mersenne31> for Standard {
     }
 }
 
+impl Serialize for Mersenne31 {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        // No need to convert to canonical.
+        serializer.serialize_u32(self.value)
+    }
+}
+
+impl<'de> Deserialize<'de> for Mersenne31 {
+    fn deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
```

Comment:
Nit: `'a` is a more idiomatic lifetime name.
```suggestion
impl<'a> Deserialize<'a> for Mersenne31 {
    fn deserialize<D: Deserializer<'a>>(d: D) -> Result<Self, D::Error> {
```